android-components/components/support/remotesettings/src/main/java/mozilla/components/support/remotesettings/RemoteSettingsClient.kt

```diff
@@ -66,54 +79,31 @@ class RemoteSettingsClient(
      */
     suspend fun write(response: RemoteSettingsResponse): RemoteSettingsResult = withContext(Dispatchers.IO) {
         try {
-            val jsonObject = JSONObject()
-            jsonObject.put("response", response)
-            AtomicFile(file).writeString { jsonObject.toString() }
+            val jsonString = Json.encodeToString(
+                SerializableRemoteSettingsResponse.serializer(),
+                response.toSerializable(),
+            )
```

Comment:
We could add another catch here for [serialization exceptions](https://kotlinlang.org/api/kotlinx.serialization/kotlinx-serialization-core/kotlinx.serialization/-serialization-exception/) if we wanted to avoid crashing.